File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.c
int lengthOfLongestSubstring(char* s) {
    int freq[128]={0};
    int right=0;
    int left=0;
    int maxlen=0;
    for(int right=0;s[right]!='\0';right++){
        freq[s[right]]++;
        while(freq[s[right]]>1){
            freq[s[left]]--;
            left++;
        }
        int len=right-left+1;
    if(len>maxlen){
        maxlen=len;
        
            }
    
    }
    return maxlen;
    
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        longest=set()
        maxlen=0
        left=0
        for right in range(len(s)):
            while s[right] in longest:
                longest.remove(s[left])
                left=left+1
            longest.add(s[right])
            maxlen=max(maxlen,len(longest))
        return maxlen
        

            
        